src/wasi.ts

```diff
@@ -255,17 +272,21 @@ export default class WASI {
           return wasi.ERRNO_BADF;
         }
       },
+
       fd_prestat_dir_name(
         fd: number,
         path_ptr: number,
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
         path_len: number,
       ): number {
         // FIXME don't ignore path_len
         if (self.fds[fd] != undefined) {
-          // @ts-ignore
-          let { ret, prestat_dir_name } = self.fds[fd].fd_prestat_dir_name();
+          const { ret, prestat_dir_name } = self.fds[fd].fd_prestat_dir_name(
+            DUMMY, //
+            DUMMY,
+          );
```

Comment:
Looks like the `fd_prestat_dir_name` function in `class Fd` has two arguments, but they are never used and all derived classes don't declare them either. Could you remove them from the `Fd.fd_prestat_dir_name` method instead?